addons/web/static/src/views/list/list_renderer.js

```diff
@@ -869,6 +869,11 @@ export class ListRenderer extends Component {
         return this.isX2Many && this.canCreate;
     }
 
+    displayDeleteIcon(record) {
+        const deleteControl = this.controls.find((control) => control.type === "delete");
```

Comment:
perf improvement proposal: find the control delete once onWillRender,, then eval it on a by record basis